jenkins-plugin/src/main/resources/org/jenkinsci/plugins/pipeline/maven/GlobalPipelineMavenConfig/help-noDataStorage.html

```diff
@@ -0,0 +1,5 @@
+<div>
+    <p>
+        Using this option means you will not store some data and disable some feature such snapshot downstream triggering (when using deploy phase)
```

Comment:
It is fine to have some inline help after somebody clicks a **?** button but a basic functional limitation needs to be more clearly notified. If there is no data storage and that means snapshot triggers do not work, then trying to run the `withMaven` step with an option requesting snapshot triggers should fail with an `AbortException`. Or if this is configured in a `Trigger` instead, then form validation for the trigger should warn you if it will not in fact work, and `withMaven` should print a warning to the upstream log that there may be configured snapshot triggers which will not be run.